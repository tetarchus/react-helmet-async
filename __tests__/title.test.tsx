import React from 'react';

import { Helmet } from '..';

import { ReactServer } from './setup/encoderFix';
import { after, before, isArray, render } from './setup/testSetup';

const NullComponent = (): null => null;

describe('Title', () => {
  beforeEach(before);
  afterEach(after);

  describe('Browser', () => {
    describe('API', () => {
      it('updates page title', () => {
        expect.assertions(1);

        render(<Helmet defer={false} defaultTitle='Fallback' title='Test Title' />);

        expect(document.title).toMatchSnapshot();
      });

      it('updates page title with multiple children', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet defer={false} title='Test Title' />
            <Helmet defer={false} title='Child One Title' />
            <Helmet defer={false} title='Child Two Title' />
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('sets title based on deepest nested component', () => {
        expect.assertions(1);
        render(
          <div>
            <Helmet defer={false} title='Main Title' />
            <Helmet defer={false} title='Nested Title' />
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('sets title using deepest nested component with a defined title', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet defer={false} title='Main Title' />
            <Helmet defer={false} />
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('uses defaultTitle if no title is defined', () => {
        expect.assertions(1);

        render(
          <Helmet
            defaultTitle='Fallback'
            defer={false}
            title=''
            titleTemplate='This is a %s of the titleTemplate feature'
          />,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('uses a titleTemplate if defined', () => {
        expect.assertions(1);

        render(
          <Helmet
            defaultTitle='Fallback'
            defer={false}
            title='Test'
            titleTemplate='This is a %s of the titleTemplate feature'
          />,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('replaces multiple title strings in titleTemplate', () => {
        expect.assertions(1);

        render(
          <Helmet
            defer={false}
            title='Test'
            titleTemplate='This is a %s of the titleTemplate feature. Another %s.'
          />,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('uses a titleTemplate based on deepest nested component', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet
              defer={false}
              title='Test'
              titleTemplate='This is a %s of the titleTemplate feature'
            />
            <Helmet
              defer={false}
              title='Second Test'
              titleTemplate='A %s using nested titleTemplate attributes'
            />
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('merges deepest component title with nearest upstream titleTemplate', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet
              defer={false}
              title='Test'
              titleTemplate='This is a %s of the titleTemplate feature'
            />
            <Helmet defer={false} title='Second Test' />
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('renders dollar characters in a title correctly when titleTemplate present', () => {
        expect.assertions(1);

        const dollarTitle = 'te$t te$$t te$$$t te$$$$t';

        render(<Helmet defer={false} title={dollarTitle} titleTemplate='This is a %s' />);

        expect(document.title).toMatchSnapshot();
      });

      it('does not encode all characters with HTML character entity equivalents', () => {
        expect.assertions(1);

        const chineseTitle = '膣膗 鍆錌雔';

        render(
          <div>
            <Helmet defer={false} title={chineseTitle} />
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('correctly renderspage title with "itemprop" prop', () => {
        expect.assertions(2);

        render(
          <Helmet
            defaultTitle='Fallback'
            defer={false}
            title='Test Title with itemProp'
            titleAttributes={{ itemprop: 'name' }}
          />,
        );

        const titleTag = document.querySelectorAll('title')[0];

        expect(document.title).toMatchSnapshot();
        expect(titleTag.getAttribute('itemprop')).toBe('name');
      });
    });

    describe('Declarative API', () => {
      it('updates page title', () => {
        expect.assertions(1);

        render(
          <Helmet defaultTitle='Fallback' defer={false}>
            <title>Test Title</title>
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('updates page title and allows children containing expressions', () => {
        expect.assertions(1);

        const someValue = 'Some Great Title';

        render(
          <Helmet defer={false}>
            <title>
              Title:
              {someValue}
            </title>
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('updates page title with multiple children', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet defer={false}>
              <title>Test Title</title>
            </Helmet>
            <Helmet defer={false}>
              <title>Child One Title</title>
            </Helmet>
            <Helmet defer={false}>
              <title>Child Two Title</title>
            </Helmet>
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('sets title based on deepest nested component', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet defer={false}>
              <title>Main Title</title>
            </Helmet>
            <Helmet defer={false}>
              <title>Nested Title</title>
            </Helmet>
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('sets title using deepest nested component with a defined title', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet defer={false}>
              <title>Main Title</title>
            </Helmet>
            <Helmet defer={false} />
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('uses defaultTitle if no title is defined', () => {
        expect.assertions(1);

        render(
          <Helmet
            defaultTitle='Fallback'
            defer={false}
            titleTemplate='This is a %s of the titleTemplate feature'
          >
            <title />
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('uses a titleTemplate if defined', () => {
        expect.assertions(1);

        render(
          <Helmet
            defaultTitle='Fallback'
            defer={false}
            titleTemplate='This is a %s of the titleTemplate feature'
          >
            <title>Test</title>
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('replaces multiple title strings in titleTemplate', () => {
        expect.assertions(1);

        render(
          <Helmet
            defer={false}
            titleTemplate='This is a %s of the titleTemplate feature. Another %s.'
          >
            <title>Test</title>
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('uses a titleTemplate based on deepest nested component', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet defer={false} titleTemplate='This is a %s of the titleTemplate feature'>
              <title>Test</title>
            </Helmet>
            <Helmet defer={false} titleTemplate='A %s using nested titleTemplate attributes'>
              <title>Second Test</title>
            </Helmet>
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('merges deepest component title with nearest upstream titleTemplate', () => {
        expect.assertions(1);

        render(
          <div>
            <Helmet defer={false} titleTemplate='This is a %s of the titleTemplate feature'>
              <title>Test</title>
            </Helmet>
            <Helmet defer={false}>
              <title>Second Test</title>
            </Helmet>
          </div>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('renders dollar characters in a title correctly when titleTemplate present', () => {
        expect.assertions(1);
        const dollarTitle = 'te$t te$$t te$$$t te$$$$t';

        render(
          <Helmet defer={false} titleTemplate='This is a %s'>
            <title>{dollarTitle}</title>
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('does not encode all characters with HTML character entity equivalents', () => {
        expect.assertions(1);

        const chineseTitle = '膣膗 鍆錌雔';

        render(
          <Helmet defer={false}>
            <title>{chineseTitle}</title>
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('correctly renderspage title with "itemprop" prop', () => {
        expect.assertions(2);

        render(
          <Helmet defaultTitle='Fallback' defer={false}>
            <title itemProp='name'>Test Title with itemProp</title>
          </Helmet>,
        );

        const titleTag = document.querySelectorAll('title')[0];

        expect(document.title).toMatchSnapshot();
        expect(titleTag.getAttribute('itemprop')).toBe('name');
      });

      it('retains existing title tag when no title tag is defined', () => {
        expect.assertions(1);

        document.head.innerHTML = `<title>Existing Title</title>`;

        render(
          <Helmet defer={false}>
            <meta name='keywords' content='stuff' />
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();
      });

      it('clears title tag if empty title is defined', () => {
        expect.assertions(2);
        render(
          <Helmet defer={false}>
            <title>Existing Title</title>
            <meta name='keywords' content='stuff' />
          </Helmet>,
        );

        expect(document.title).toMatchSnapshot();

        render(
          <Helmet defer={false}>
            <title />
            <meta name='keywords' content='stuff' />
          </Helmet>,
        );

        expect(document.title).toBe('');
      });
    });
  });

  describe('Server', () => {
    describe('API', () => {
      it('provides initial values if no state is found', () => {
        expect.assertions(3);

        const { context } = render(<NullComponent />, true);
        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.meta).toBeDefined();
        expect(head?.meta.toString).toBeDefined();
        expect(head?.meta.toString()).toBe('');
      });

      it('encodes special characters in title', () => {
        expect.assertions(3);

        const { context } = render(
          <Helmet defer={false} title='Dangerous <script> include' />,
          true,
        );
        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('opts out of string encoding', () => {
        expect.assertions(3);

        const { context } = render(
          <Helmet
            defer={false}
            encodeSpecialCharacters={false}
            title={"This is text and & and '."}
          />,
          true,
        );
        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('renders title as React component', () => {
        expect.assertions(6);

        const { context } = render(
          <Helmet defer={false} title='Dangerous <script> include' />,
          true,
        );
        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toComponent).toBeDefined();

        const titleComponent = head?.title.toComponent() ?? [];

        expect(titleComponent).toStrictEqual(isArray);
        expect(titleComponent).toHaveLength(1);

        for (const title of titleComponent) {
          expect(title).toStrictEqual(expect.objectContaining({ type: 'title' }));
        }

        const view = ReactServer.renderToStaticMarkup(<>{titleComponent}</>);

        expect(view).toMatchSnapshot();
      });

      it('renders title with itemprop name as React component', () => {
        expect.assertions(6);

        const { context } = render(
          <Helmet
            defer={false}
            title='Title with Itemprop'
            titleAttributes={{ itemprop: 'name' }}
          />,
          true,
        );
        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toComponent).toBeDefined();

        const titleComponent = head?.title.toComponent() ?? [];

        expect(titleComponent).toStrictEqual(isArray);
        expect(titleComponent).toHaveLength(1);

        for (const title of titleComponent) {
          expect(title).toStrictEqual(expect.objectContaining({ type: 'title' }));
        }

        const view = ReactServer.renderToStaticMarkup(<>{titleComponent}</>);

        expect(view).toMatchSnapshot();
      });

      it('renders title tag as string', () => {
        expect.assertions(3);

        const { context } = render(
          <Helmet defer={false} title='Dangerous <script> include' />,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('renders title with itemprop name as string', () => {
        expect.assertions(3);

        const { context } = render(
          <Helmet
            defer={false}
            title='Title with Itemprop'
            titleAttributes={{ itemprop: 'name' }}
          />,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();

        const titleString = head?.title.toString();

        expect(titleString).toMatchSnapshot();
      });

      it('does not encode all characters with HTML character entity equivalents', () => {
        expect.assertions(3);

        const chineseTitle = '膣膗 鍆錌雔';

        const { context } = render(
          <div>
            <Helmet defer={false} title={chineseTitle} />
          </div>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });
    });

    describe('Declarative API', () => {
      it('encodes special characters in title', () => {
        expect.assertions(3);
        const { context } = render(
          <Helmet defer={false}>
            <title>{`Dangerous <script> include`}</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('opts out of string encoding', () => {
        expect.assertions(3);

        const { context } = render(
          <Helmet encodeSpecialCharacters={false}>
            {/* Testing unescaped characters */}
            {/* eslint-disable-next-line react/no-unescaped-entities */}
            <title>This is text and & and '.</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('renders title as React component', () => {
        expect.assertions(6);

        const { context } = render(
          <Helmet defer={false}>
            <title>{`Dangerous <script> include`}</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toComponent).toBeDefined();

        const titleComponent = head?.title.toComponent() ?? [];

        expect(titleComponent).toStrictEqual(isArray);
        expect(titleComponent).toHaveLength(1);

        for (const title of titleComponent) {
          expect(title).toStrictEqual(expect.objectContaining({ type: 'title' }));
        }

        const view = ReactServer.renderToStaticMarkup(<>{titleComponent}</>);

        expect(view).toMatchSnapshot();
      });

      it('renders title with itemprop name as React component', () => {
        expect.assertions(6);

        const { context } = render(
          <Helmet defer={false}>
            <title itemProp='name'>Title with Itemprop</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toComponent).toBeDefined();

        const titleComponent = head?.title.toComponent() ?? [];

        expect(titleComponent).toStrictEqual(isArray);
        expect(titleComponent).toHaveLength(1);

        for (const title of titleComponent) {
          expect(title).toStrictEqual(expect.objectContaining({ type: 'title' }));
        }

        const view = ReactServer.renderToStaticMarkup(<>{titleComponent}</>);

        expect(view).toMatchSnapshot();
      });

      it('renders title tag as string', () => {
        expect.assertions(3);

        const { context } = render(
          <Helmet defer={false}>
            <title>{'Dangerous <script> include'}</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('renders title and allows children containing expressions', () => {
        expect.assertions(3);

        const someValue = 'Some Great Title';

        const { context } = render(
          <Helmet defer={false}>
            <title>
              Title:
              {someValue}
            </title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('renders title with itemprop name as string', () => {
        expect.assertions(3);

        const { context } = render(
          <Helmet defer={false}>
            <title itemProp='name'>Title with Itemprop</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();

        const titleString = head?.title.toString();

        expect(titleString).toMatchSnapshot();
      });

      it('does not encode all characters with HTML character entity equivalents', () => {
        expect.assertions(3);

        const chineseTitle = '膣膗 鍆錌雔';

        const { context } = render(
          <div>
            <Helmet defer={false}>
              <title>{chineseTitle}</title>
            </Helmet>
          </div>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });
    });

    describe('RenderStatic', () => {
      it('html encodes title', () => {
        expect.assertions(3);
        const { context } = render(
          <Helmet defer={false}>
            <title>{`Dangerous <script> include`}</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toString).toBeDefined();
        expect(head?.title.toString()).toMatchSnapshot();
      });

      it('renders title as React component', () => {
        expect.assertions(6);

        const { context } = render(
          <Helmet defer={false}>
            <title>{`Dangerous <script> include`}</title>
          </Helmet>,
          true,
        );

        const head = 'helmet' in context ? context.helmet : null;

        expect(head?.title).toBeDefined();
        expect(head?.title.toComponent).toBeDefined();

        const titleComponent = head?.title.toComponent() ?? [];

        expect(titleComponent).toStrictEqual(isArray);
        expect(titleComponent).toHaveLength(1);

        for (const title of titleComponent) {
          expect(title).toStrictEqual(expect.objectContaining({ type: 'title' }));
        }

        const view = ReactServer.renderToStaticMarkup(<>{titleComponent}</>);

        expect(view).toMatchSnapshot();
      });
    });
  });
});
