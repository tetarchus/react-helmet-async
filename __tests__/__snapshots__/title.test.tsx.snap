// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Browser API correctly renderspage title with "itemprop" prop 1`] = `"Test Title with itemProp"`;

exports[`Title Browser API does not encode all characters with HTML character entity equivalents 1`] = `"膣膗 鍆錌雔"`;

exports[`Title Browser API merges deepest component title with nearest upstream titleTemplate 1`] = `"Second Test"`;

exports[`Title Browser API renders dollar characters in a title correctly when titleTemplate present 1`] = `"This is a te$t te$$t te$$$t te$$$$t"`;

exports[`Title Browser API replaces multiple title strings in titleTemplate 1`] = `"This is a Test of the titleTemplate feature. Another Test."`;

exports[`Title Browser API sets title based on deepest nested component 1`] = `"Nested Title"`;

exports[`Title Browser API sets title using deepest nested component with a defined title 1`] = `""`;

exports[`Title Browser API updates page title 1`] = `"Test Title"`;

exports[`Title Browser API updates page title with multiple children 1`] = `"Child Two Title"`;

exports[`Title Browser API uses a titleTemplate based on deepest nested component 1`] = `"A Second Test using nested titleTemplate attributes"`;

exports[`Title Browser API uses a titleTemplate if defined 1`] = `"This is a Test of the titleTemplate feature"`;

exports[`Title Browser API uses defaultTitle if no title is defined 1`] = `"This is a of the titleTemplate feature"`;

exports[`Title Browser Declarative API clears title tag if empty title is defined 1`] = `"Existing Title"`;

exports[`Title Browser Declarative API correctly renderspage title with "itemprop" prop 1`] = `"Test Title with itemProp"`;

exports[`Title Browser Declarative API does not encode all characters with HTML character entity equivalents 1`] = `"膣膗 鍆錌雔"`;

exports[`Title Browser Declarative API merges deepest component title with nearest upstream titleTemplate 1`] = `"Second Test"`;

exports[`Title Browser Declarative API renders dollar characters in a title correctly when titleTemplate present 1`] = `"This is a te$t te$$t te$$$t te$$$$t"`;

exports[`Title Browser Declarative API replaces multiple title strings in titleTemplate 1`] = `"This is a Test of the titleTemplate feature. Another Test."`;

exports[`Title Browser Declarative API retains existing title tag when no title tag is defined 1`] = `""`;

exports[`Title Browser Declarative API sets title based on deepest nested component 1`] = `"Nested Title"`;

exports[`Title Browser Declarative API sets title using deepest nested component with a defined title 1`] = `""`;

exports[`Title Browser Declarative API updates page title 1`] = `"Test Title"`;

exports[`Title Browser Declarative API updates page title and allows children containing expressions 1`] = `"Title:Some Great Title"`;

exports[`Title Browser Declarative API updates page title with multiple children 1`] = `"Child Two Title"`;

exports[`Title Browser Declarative API uses a titleTemplate based on deepest nested component 1`] = `"A Second Test using nested titleTemplate attributes"`;

exports[`Title Browser Declarative API uses a titleTemplate if defined 1`] = `"This is a Test of the titleTemplate feature"`;

exports[`Title Browser Declarative API uses defaultTitle if no title is defined 1`] = `"Fallback"`;

exports[`Title Server API does not encode all characters with HTML character entity equivalents 1`] = `"<title data-rh=\\"true\\">膣膗 鍆錌雔</title>"`;

exports[`Title Server API encodes special characters in title 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;

exports[`Title Server API opts out of string encoding 1`] = `"<title data-rh=\\"true\\">This is text and & and '.</title>"`;

exports[`Title Server API renders title as React component 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;

exports[`Title Server API renders title tag as string 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;

exports[`Title Server API renders title with itemprop name as React component 1`] = `"<title data-rh=\\"true\\" itemProp=\\"name\\">Title with Itemprop</title>"`;

exports[`Title Server API renders title with itemprop name as string 1`] = `"<title data-rh=\\"true\\" itemprop=\\"name\\">Title with Itemprop</title>"`;

exports[`Title Server Declarative API does not encode all characters with HTML character entity equivalents 1`] = `"<title data-rh=\\"true\\">膣膗 鍆錌雔</title>"`;

exports[`Title Server Declarative API encodes special characters in title 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;

exports[`Title Server Declarative API opts out of string encoding 1`] = `"<title data-rh=\\"true\\">This is text and & and '.</title>"`;

exports[`Title Server Declarative API renders title and allows children containing expressions 1`] = `"<title data-rh=\\"true\\">Title:Some Great Title</title>"`;

exports[`Title Server Declarative API renders title as React component 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;

exports[`Title Server Declarative API renders title tag as string 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;

exports[`Title Server Declarative API renders title with itemprop name as React component 1`] = `"<title data-rh=\\"true\\" itemProp=\\"name\\">Title with Itemprop</title>"`;

exports[`Title Server Declarative API renders title with itemprop name as string 1`] = `"<title data-rh=\\"true\\" itemprop=\\"name\\">Title with Itemprop</title>"`;

exports[`Title Server RenderStatic html encodes title 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;

exports[`Title Server RenderStatic renders title as React component 1`] = `"<title data-rh=\\"true\\">Dangerous &lt;script&gt; include</title>"`;
